### PR TITLE
Jump and vehicle start/stop improvements

### PR DESCRIPTION
**Pull request by paulkocialkowski**
This adds a few improvements that fix issues that were slightly annoying when playing the game:
- Jumps did not get inertia from the previous movement
- Cars were very slow to start and stop, which was very noticeable when going from forward to reverse

### CODE REVIEW

File: rwengine/src/objects/CharacterObject.cpp [hunk 1]
```diff
@@ -35,6 +35,7 @@ CharacterObject::CharacterObject(GameWorld* engine, const glm::vec3& pos,
     , m_look(0.f, glm::half_pi<float>())
     , running(false)
     , jumped(false)
+    , jumpAnimation(NULL)
```
Comment (ShFil119) [inline rwengine/src/objects/CharacterObject.cpp]: Instead of NULL you should use nullptr.

File: rwengine/src/objects/CharacterObject.cpp [hunk 2]
```diff
@@ -490,6 +504,12 @@ void CharacterObject::jump() {
         physCharacter->jump(btVector3(0.f, 0.f, 0.f));
 #endif
         jumped = true;
+        jumpAnimation = animator->getAnimation(AnimIndexMovement);
+
+        // There is no kinetic energy left after a jump
+        if (jumpAnimation == animations->animation(AnimCycle::JumpLand))
```
Comment (ShFil119) [inline rwengine/src/objects/CharacterObject.cpp]: After character lands there's weird feeling, like character's body lose its ability move horizontally.

File: rwengine/src/objects/CharacterObject.cpp [hunk 3]
```diff
@@ -201,44 +248,11 @@ glm::vec3 CharacterObject::updateMovementAnimation(float dt) {
...
-            glm::vec3 d = (b - a);
-            animTranslate.y += d.y;
-
-            // Kludge: Drop y component of root bone
-            auto t = glm::vec3(root->getTransform()[3]);
-            t.y = 0.f;
-            root->setTranslation(t);
-        }
-    }
+    if (jumped == true && jumpAnimation != NULL)
```
Comment (ShFil119) [inline rwengine/src/objects/CharacterObject.cpp]: Also here.

File: rwengine/src/objects/CharacterObject.cpp (hunk 1 above)
Comment (paulkocialkowski) [reply rwengine/src/objects/CharacterObject.cpp]: Ah thanks, it shows that I've used to C over C++...

**Comment by dracc** [comment]
Is this PR dead or do you plan to fix the pointed out issues, @paulkocialkowski ? :+1:

**Comment by darkf** [comment]
@paulkocialkowski Thanks for your contribution.

Since it seems there are some unresolved issues (+ it needs a rebase), I'm going to close this for now to avoid confusion (and taking space in the queue), however you are more than welcome to open a new PR when/if you fix those issues. Thanks. 👍

**Pull request closed without merging**